### DISCUSSION TITLE
Stray file with vcpkg_extract_source_archive_ex and NO_REMOVE_ONE_LEVEL

### DISCUSSION THREAD

**Discussion by PhilLab** [microsoft/vcpkg#19145] [Ideas]
**Describe the bug**

**Environment**
- OS: Windows
- Vcpkg package management program version 2019.09.12

**To Reproduce**
Use a port file with the `NO_REMOVE_ONE_LEVEL` of `vcpkg_extract_source_archive_ex` . Example:

    vcpkg_download_distfile(ARCHIVE
        URLS "https://github.com/Genymobile/scrcpy/releases/download/v1.10/scrcpy-win64-v1.10.zip"
        FILENAME "scrcpy-win64-v1.10.zip"
        SHA512 04fccaffb83ea4e07dc50fe49ca2de03d1ff53ca8b96e2de72569e0d6173b861cdab12c5a50b77776fd7d701338029c52f124c95ba84928b978788ad8ac1c604
    )
    vcpkg_extract_source_archive_ex(
        OUT_SOURCE_PATH ARCHIVE_EXTRACTED
        ARCHIVE ${ARCHIVE}
        REF "scrcpy"
        NO_REMOVE_ONE_LEVEL
    )
    file(INSTALL ${ARCHIVE_EXTRACTED}
        DESTINATION ${CURRENT_PACKAGES_DIR}/tools/${PORT}/
        RENAME scrcpy
    )

the installation will contain a `scrcpy-win64-v1.10.zip.extracted` file.

One can workaround this by modifying the file command like this:

    file(INSTALL ${ARCHIVE_EXTRACTED}
        DESTINATION ${CURRENT_PACKAGES_DIR}/tools/${PORT}/
        RENAME scrcpy
        PATTERN "**/*.extracted" EXCLUDE
    )

**Expected behavior**
No stray .extracted file appears

**Comment by vpaladino778** [comment]
I'll take a look and see if i can come up with a fix for this

**Comment by vpaladino778** [comment]
Looks like this issue persists even when not using `NO_REMOVE_ONE_LEVEL`. Is it possible that it's this specific package that's causing the issue?

**Comment by PhilLab** [comment]
@vpaladino778 The port file is the one above. I also saw this with several other packages so I suppose this is a systemic issue, see this screenshot from my build tree:

![image](https://user-images.githubusercontent.com/1710255/67484208-df4de800-f667-11e9-9b9f-8e989c686d82.png)
